code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/lb/LoadBalancerHostValidationFilter.java

```diff
@@ -0,0 +1,71 @@
+package io.cattle.platform.iaas.api.filter.lb;
+
+import io.cattle.platform.core.constants.LoadBalancerConstants;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class LoadBalancerHostValidationFilter extends AbstractDefaultResourceManagerFilter {
+    
+    GenericMapDao mapDao;
+
+    private static final Map<String, Boolean> actions;
+    static
+    {
+        actions = new HashMap<>();
+        actions.put(LoadBalancerConstants.ACTIONT_ADD_HOST.toLowerCase(), true);
+        actions.put(LoadBalancerConstants.ACTIONT_REMOVE_HOST.toLowerCase(), false);
+    }
+    
+    @Override
+    public String[] getTypes() {
+        return new String[] { LoadBalancerConstants.OBJECT_LB };
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { LoadBalancer.class };
+    }
+
+    @Override
+    public Object resourceAction(String type, ApiRequest request, ResourceManager next) {
+        if (actions.containsKey(request.getAction())) {
+            Map<String,Object> data = CollectionUtils.toMap(request.getRequestObject());
+            validateAction(Long.valueOf(request.getId()), data, request.getAction());
+        }
+
+        return super.resourceAction(type, request, next);
+    }
+
+    private void validateAction(long id, Map<String, Object> data, String action) {
+        Long hostId = (Long) data.get(LoadBalancerConstants.FIELD_LB_HOST_ID);
+
+        if (actions.get(action)) {
+            if (mapDao.findNonRemoved(LoadBalancerHostMap.class, Host.class, hostId, LoadBalancer.class, id) != null) {
+                ValidationErrorCodes.throwValidationError(ValidationErrorCodes.NOT_UNIQUE,
+                        LoadBalancerConstants.FIELD_LB_HOST_ID);
+            }
+        } else {
+            if (mapDao.findToRemove(LoadBalancerHostMap.class, Host.class, hostId, LoadBalancer.class, id) == null) {
+                ValidationErrorCodes.throwValidationError(ValidationErrorCodes.INVALID_OPTION,
+                        LoadBalancerConstants.FIELD_LB_HOST_ID);
+            }
+        }
+    }
+
+    @Inject
```

Comment:
Side note.  Although the current cattle code base doesn't do this we've decided going forward that we can just add @Inject to class member fields and save all those useful bytes wasted on accessor methods for bean properties.  Just imagine if we refactored the whole code base like that, we'd probably have half the lines :)